### ISSUE TITLE
Checking "If condition" error.

### ISSUE THREAD

**Issue by koftikes** [phpstan/phpstan-strict-rules#42]
Exampe:
```php
<?php declare(strict_types=1);

$arr = [];

if ($arr) {
    echo 'some';
}

if (!empty($arr)) {
    echo 'some';
}
```

If I have some array in return and check it in "if condition" I got next errors:
  5      Only booleans are allowed in an if condition, array given.     
  9      Construct empty() is not allowed. Use more strict comparison.  

So my question is: How it should be fixed?

**Comment by ondrejmirtes** [comment]
```php
<?php declare(strict_types=1);

$arr = [];

if (count($arr) > 0) {
    echo 'some';
}

if (count($arr) > 0) {
    echo 'some';
}
```

The problem with `empty` is that it hides undefined variables and many types are considered empty, which would suprise you.
